pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go

```diff
@@ -42,45 +60,81 @@ func validateEnrichMetrics(metrics []MetricsConfig) []string {
 			}
 			for i := range metricConfig.MetricTags {
 				metricTag := &metricConfig.MetricTags[i]
-				errors = append(errors, validateEnrichMetricTag(metricTag, metricConfig)...)
+				errors = append(errors, validateEnrichMetricTag(metricTag)...)
 			}
 		}
 	}
 	return errors
 }
 
-func validateEnrichSymbol(symbol *SymbolConfig, metricConfig *MetricsConfig) []string {
+// validateEnrichMetadata will validate MetadataConfig and enrich it.
+func validateEnrichMetadata(metadata MetadataConfig) []string {
+	var errors []string
+	for resName := range metadata {
+		_, isValidRes := validMetadataResources[resName]
+		if !isValidRes {
+			errors = append(errors, fmt.Sprintf("invalid resource: %s", resName))
+		} else {
+			res := metadata[resName]
+			for fieldName := range metadata[resName].Fields {
+				_, isValidField := validMetadataResources[resName][fieldName]
+				if !isValidField {
+					errors = append(errors, fmt.Sprintf("invalid resource (%s) field: %s", resName, fieldName))
+					continue
+				}
+				field := res.Fields[fieldName]
+				if field.Value == "" && field.Symbol.OID == "" {
+					errors = append(errors, fmt.Sprintf("field `%s`: value or symbol cannot be both empty", fieldName))
+				}
+				if field.Value != "" && field.Symbol.OID != "" {
+					errors = append(errors, fmt.Sprintf("field `%s`: value or symbol cannot be both defined", fieldName))
+				}
+				if field.Symbol.OID != "" {
+					errors = append(errors, validateEnrichSymbol(&field.Symbol)...)
+				}
+				res.Fields[fieldName] = field
```

Comment:
Why do you assign `field` back to `res.Fields[fieldName]`, after getting it at line 85?

Reply:
That's because `field` is a copy of the original one (`field := res.Fields[fieldName]`), without that, the changes won't be reflected in `metadata`.